packages/internal/src/generate/graphqlCodeGen.ts

```diff
@@ -188,16 +214,12 @@ export const getResolverFnType = () => {
     return `(
       args: TArgs,
       obj?: { root: TParent; context: TContext; info: GraphQLResolveInfo }
-    ) => TResult extends PromiseLike<infer TResultAwaited>
-      ? Promise<Partial<TResultAwaited>>
-      : Promise<Partial<TResult>> | Partial<TResult>;`
+    ) => TResult | Promise<TResult>`
```

Comment:
This return Resolver's ReturnType to the default. I introduced this in strict-mode and have subsquently realised this is a terrible idea.